homeassistant/components/zha/config_flow.py

```diff
@@ -648,9 +648,7 @@ async def async_step_hardware(
         radio_type = self._radio_mgr.parse_radio_type(discovery_data["radio_type"])
 
         try:
-            device_settings = radio_type.controller.SCHEMA_DEVICE(
-                discovery_data["port"]
-            )
+            device_settings = SCHEMA_DEVICE(discovery_data["port"])
```

Comment:
It's not good that the config entry data structure comes directly from a 3rd party library. There should be an explicit translation here in the integration so an update to the 3rd party library can't on its own change the structure of the config entry data created.

Reply:
Done. I've made the schema explicit within ZHA and use it when creating a new config entry.